hoot-core/src/main/cpp/hoot/core/conflate/highway/HighwayMatchCreator.cpp

```diff
@@ -123,14 +125,17 @@ class HighwayMatchVisitor : public ElementVisitor
   static HighwayMatch* createMatch(const ConstOsmMapPtr& map,
     shared_ptr<HighwayClassifier> classifier,
     shared_ptr<SublineStringMatcher> sublineMatcher,
-    ConstMatchThresholdPtr threshold, ConstElementPtr e1, ConstElementPtr e2)
+    ConstMatchThresholdPtr threshold,
+    shared_ptr<TagAncestorDifferencer> tagAncestorDiff,
+    ConstElementPtr e1, ConstElementPtr e2)
   {
     HighwayMatch* result = 0;
 
     if (e1 && e2 &&
         e1->getStatus() != e2->getStatus() && e2->isUnknown() &&
         OsmSchema::getInstance().isLinearHighway(e1->getTags(), e1->getElementType()) &&
-        OsmSchema::getInstance().isLinearHighway(e2->getTags(), e2->getElementType()))
+        OsmSchema::getInstance().isLinearHighway(e2->getTags(), e2->getElementType()) &&
+        tagAncestorDiff->diff(map, e1, e2) <= ConfigOptions().getHighwayMaxEnumDiff())
```

Comment:
You should read this once at the beginning and re-use it. `ConfigOptions().getHighwayMaxEnumDiff()`

That call can be a little slow.